src/analytics/src/main/scala/osmesa/analytics/oneoffs/Roads.scala

```diff
@@ -0,0 +1,64 @@
+package osmesa.analytics.oneoffs
+
+import scala.util.{Try, Success, Failure}
+
+import cats.implicits._
+import com.monovore.decline._
+import geotrellis.vector.{Feature, Line}
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql._
+import vectorpipe._
+
+object Roads extends CommandApp(
+
+  name   = "road-changes",
+  header = "How many kilometers of roads changed?",
+  main   = {
+
+    val orcO = Opts.option[String]("orc", help = "Location of the ORC file to process.")
+
+    orcO.map { orc =>
+
+      /* Settings compatible for both local and EMR execution */
+      val conf = new SparkConf()
+        .setIfMissing("spark.master", "local[*]")
+        .setAppName("road-changes")
+        .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+        .set("spark.kryo.registrator", classOf[geotrellis.spark.io.kryo.KryoRegistrator].getName)
+
+      implicit val ss: SparkSession = SparkSession.builder
+        .config(conf)
+        .enableHiveSupport
+        .getOrCreate
+
+      /* Silence the damn INFO logger */
+      Logger.getRootLogger().setLevel(Level.ERROR)
+
+      (Try(ss.read.orc(orc)) >>= Analysis.roads) match {
+        case Failure(e) => println(e)
+        case Success(d) => println(s"${d} kilometers of roads were changed.")
+      }
+
+      ss.stop()
+
+    }
+  }
+)
+
+object Analysis {
+
+  /** How many kilometers of road changed in all the Ways present in the given DataFrame? */
+  def roads(data: DataFrame)(implicit ss: SparkSession): Try[Double] = {
+
+    Try(osm.fromDataFrame(data)).map { case (nodes, ways, relations) =>
+      val roadsOnly: RDD[(Long, osm.Way)] = ways.filter(_._2.data.tagMap.contains("road"))
```

Comment:
My sense is that if this filter gets performed prior to parsing the DF into case classes, it'll be more performant.

Reply:
Oh absolutely - I was going for the more immediately available approach first. On a Friday afternoon it wasn't immediately obvious to me what the SparkSQL would've been to say:

> Take this Dataframe and give me a reduced form of it that contains all Nodes and Ways, such that the `TAGS` map in each way contains the key `road`.

Reply:
Gotcha, that makes sense. The SQL isnt directly obvious to me as well, might take `udf` to do it. When you get there, happy to talk through it more